dice_ml/explainer_interfaces/dice_KD.py

```diff
@@ -82,7 +82,10 @@ def _generate_counterfactuals(self, query_instance, total_CFs, desired_range=Non
 
         # Prepares user defined query_instance for DiCE.
         query_instance_orig = query_instance.copy()
-        query_instance = self.data_interface.prepare_query_instance(query_instance=query_instance)
+        query_instance_orig = self.data_interface.prepare_query_instance(
```

Comment:
Sorry, I don't understand the fix here? Could you elaborate?

Reply:
The code was using two local variables, `query_instance` and `query_instance_orig`. The second variable was not passed through `prepare_query_instance`, hence the categorical features were not properly encoded.

Reply:
Thanks for clarifying